homeassistant/components/fivem/config_flow.py

```diff
@@ -0,0 +1,71 @@
+"""Config flow for FiveM integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from fivem import FiveM, FiveMServerOfflineError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_PORT = 30120
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): cv.string,
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> str | None:
+    """Validate the user input allows us to connect."""
+
+    fivem = FiveM(data[CONF_HOST], data[CONF_PORT])
+    info = await fivem.get_info_raw()
+
+    return info.get("vars")["gamename"]
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for FiveM."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            gamename = await validate_input(self.hass, user_input)
+
+            if gamename is None or gamename != "gta5":
```

Comment:
Should it raise a different error (and show a different message) if the gamename is not `gta5`?

Reply:
It will add better error messages to the config flow